### ISSUE TITLE
Page Break Issues

### ISSUE THREAD

**Issue by rkarina** [ArthurHub/HTML-Renderer#49]
Hi all, 

I've added a page-break-after: always attribute to a div element in the HTML string/snippet, to separate content and paging. The PDF converted should produce multiple pages, however, it does not. Is anyone able to see whether page break are handled, especially page-break-after CSS?

Thanks in advanced!

Regards, 

Rahmadhy

**Comment by gyfke** [comment]
Hi Rahmadhy

At the moment property page-break-after is not supported. As well as page-break-before

**Comment by rkarina** [comment]
Hi,

Is there any way for page breaking to be included in the HTML Renderer?

**Comment by rkarina** [comment]
Are you able to develop the source code to support page-break-after?

**Comment by rkarina** [comment]
Hi gyfke, 

Is there a way to include the handling of page-break-after and page-break-before properties?

**Comment by rkarina** [comment]
Hi Andrey,

Are you willing to be able to handle the page-break-after property for a commission? I'm in urgent need of the page break handling to work properly.

Please reply if interested and contact me at ak@rms.com.au.

Regards,

Rahmadhy

**Comment by Grasher134** [comment]
Any news on this feature? Adding forced page breaks after tables is necessary for my project.

**Comment by rkarina** [comment]
I haven't received a reply yet from Andrey. I've contacted the main
contributor (Arthur) and he has responded by not being able to support the
forced page break functionality.

**Comment by Grasher134** [comment]
I made my own implementation by cutting big htmls into small ones, creating several pdf files from them and adding pages from these files into 1 pdf. Not an elegant approach, but the only one I could use for now.

>  I haven't received a reply yet from Andrey. I've contacted the main
> contributor (Arthur) and he has responded by not being able to support the
> forced page break functionality.

**Comment by rkarina** [comment]
Sounds like a good solution. Is there any way you could guide me to getting
that done? I need to implement that for my project also.

**Comment by Grasher134** [comment]
```
 PdfDocument doc = new PdfDocument();

 foreach (NBElement htmlString in arrayOfStrings)
 {
  PdfDocument tempDoc =  getPdfDocFrom(htmlString.GetString());
 addPagesToPdf(ref doc, tempDoc);
 }

```

 ```
private PdfDocument getPdfDocFrom (string htmlString)
        {
            PdfGenerateConfig config = new PdfGenerateConfig();
            config.PageOrientation = PdfSharp.PageOrientation.Landscape;
            config.PageSize = PdfSharp.PageSize.A4;

            PdfDocument doc = PdfGenerator.GeneratePdf(htmlString, config);

            return doc;
        }

```

```
  
 private static void addPagesToPdf(ref PdfDocument mainDoc, PdfDocument sourceDoc)

        {

            MemoryStream tempMemoryStream = new MemoryStream();
            sourceDoc.Save(tempMemoryStream, false);

            PdfDocument openedDoc = PdfReader.Open(tempMemoryStream, PdfDocumentOpenMode.Import);
            foreach (PdfPage page in openedDoc.Pages)
            {

                mainDoc.AddPage(page);
                
            }
        }


```

**Comment by Grasher134** [comment]
How you cut your html - is up to you.

**Comment by rkarina** [comment]
Yes, I set the page size by the configuration options they offer (A4, A5,
etc). However, it does not force a page break when I add a div element with
the page-break-after: always CSS property. Is there any way to handle this?

Example:

"<p>Page One</p>
<div style="page-break-after: always;"><span style="display: none;">some
text</span></div>
<p>Page Two</p>
"

This should show a PDF document with two pages and Page One/Two on the top
of each page.

**Comment by Grasher134** [comment]
I told you, make them 2 different html strings:
string a = "<p>Page One</p>
<div style="page-break-after: always;"><span style="display: none;">some
text</span></div>"

string b = "<p>Page Two</p>"

Insert them into some array of strings and then just use my code. You will get what you need. (Oh and you can get rid of the "page-break-after: always;" as it doesn't work here).

**Comment by rkarina** [comment]
Oh okay, thank you so much Azimov. I'll let you know if I need further
assistance. Appreciate your help! :)
